src/utils/index.js

```diff
@@ -24,3 +24,25 @@ export class FastFillStyle {
     }
   }
 }
+
+/**
+ * Perform a simple shallow object equality check.
+ */
+export function objectShallowEquals(a: Object, b: Object): boolean {
+  let aLength = 0;
+  let bLength = 0;
```

Comment:
I didn't go with this approach since it adds two array allocations. As far as I'm aware, there is no allocation-free approach to getting object length, otherwise I would do this approach.